packages/flutter_tools/lib/src/commands/channel.dart

```diff
@@ -75,11 +77,20 @@ class ChannelCommand extends FlutterCommand {
           return null;
         }
         seenChannels.add(branchName);
-        if (branchName == currentBranch)
-          return '* $branchName';
-        if (!branchName.startsWith('HEAD ') &&
-            (showAll || FlutterVersion.officialChannels.contains(branchName)))
-          return '  $branchName';
+
+        if (!branchName.startsWith('HEAD ')) {
```

Comment:
Since you're changing the order of the output here, it would be more straightforward if you collect all output into a List<String>, processed that list, and then printed it out

Reply:
Yes, at the end the complexity i was facing was the fact that i needed a `reduceFunction` instead of a `mapFunction` to sort them before to print them out, but actually i found your proposed approach way more clear than using the `counter`